hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -353,6 +353,8 @@ private static EthTxData populateLegacyEthTxData(RLPItem rlpItem, byte[] rawTx)
         byte recId = vBI.testBit(0) ? (byte) 0 : 1;
         if (vBI.compareTo(BigInteger.valueOf(34)) > 0) {
             chainId = vBI.subtract(BigInteger.valueOf(35)).shiftRight(1).toByteArray();
+        } else if (vBI.compareTo(BigInteger.valueOf(27)) == 0 || vBI.compareTo(BigInteger.valueOf(28)) == 0) {
+            chainId = new byte[0];
```

Comment:
Don't know if this is correct or not (outside of my domain knowledge at this time) but in any event certainly needs a comment to explain the circumstances.

(Also, is an empty array for `chainId` correct, instead of, say, a 2-byte array of `0`?  Not sure how this is used, or why empty array is better than null.)

Reply:
later on check [here](https://github.com/hashgraph/hedera-services/blob/11401-eip155-support/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/HevmTransactionFactory.java#L281) the two values if they match and null won't be a good option

Reply:
1) Can't we then have a named constant for the 0-length byte array? 
2) Similarly, are there any names we can use for the magic numbers `27` and 35`?
3) Can any of this be abstracted behind (short 1-line) methods with reasonable names?

Goal in all these questions is to arrange things so that in the future, when someone is trying to read this, they aren't working down at the level of whether a byte array is 0-length or `null` or something else, or whether the number is supposed to be `27` or maybe `26` or even `28` but are using terms that will be familiar to them (assuming they're familiar with the Ethereum semantics)?

Reply:
1. `UNPROTECTED_TRANSACTION` would be a good name as it will communicate the effect as well.
2. There are no names for these magic numbers, even in the yellow paper and EIP-155 where they are defined.
3. I view this class as the abstraction. You pass in transaction bytes and out comes `recId` and `chainId` as fields in the object.  There is literally no where else this should need to be know.

Reply:
Agree with sentiments above
Here's a combination of them for inspiration
```suggestion
     ...
     // Both Ethereum  yellow paper and https://eips.ethereum.org/EIPS/eip-155 note support for v = 27|28 cases in unprotected transaction cases
     static final BigInteger LEGACY_V_BYTE_SIGNATURE_0 = BigInteger.valueOf(27);
     static final BigInteger LEGACY_V_BYTE_SIGNATURE_1 = BigInteger.valueOf(28);
        ...
        } else if (isVBIFromUnprotectedTransaction(vBI)) {
            chainId = new byte[0];
        ....
        private static boolean isVBIOfUnprotectedTransactionType(BigInteger vBI) {
            vBI.compareTo(LEGACY_V_BYTE_SIGNATURE_0) == 0 || vBI.compareTo(LEGACY_V_BYTE_SIGNATURE_1) == 0
        }
```

Reply:
TY Nana, this is in fact what I had in mind to make this code more clear in the future.  (And also I see that comments have been added to point to EIP-155 where they're needed.)